nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -70,36 +70,42 @@ class GroupToRank:
     operations: List[NNCFNode]
 
 
-class Ranker(ABC):
+class Ranker:
     """
     Encapsulates logic to rank groups of quantizers.
     """
 
     def __init__(
-        self,
-        ranking_subset_size: int,
-        ranking_fn: Callable[[Any, Any], float],
-        dataset: Dataset,
-        algo_backend: AccuracyControlAlgoBackend,
+        self, ranking_subset_size: int, dataset: Dataset, algo_backend: AccuracyControlAlgoBackend, evaluator: Evaluator
     ):
         """
         :param ranking_subset_size: The number of data items that will be selected from
             the dataset to rank groups of quantizers. The `len(dataset)` data items will
             be selected if `ranking_subset_size` parameter is greater than the number of
             elements in the dataset.
-        :param ranking_fn: A function that compares values returned by
-            `_collect_values_for_each_item()` for initial and quantized models.
         :param dataset: Dataset for the ranking process.
         :param algo_backend: The `AccuracyControlAlgoBackend` algo backend.
+        :param evaluator: Evaluator to validate model.
         """
         self._ranking_subset_size = ranking_subset_size
-        self._ranking_fn = ranking_fn
         self._dataset = dataset
         self._algo_backend = algo_backend
+        self._evaluator = evaluator
         # We don't need to re-calculate values for the initial model
         # because they don't change. So use this attribute to store
         # them to improve execution time.
-        self._ref_values = None
+        self._reference_values_for_each_item = None
+
+        # A `ranking_fn()` is a function that compares values returned
+        # by `Evaluator.collect_values_for_each_item()` method for
+        # initial and quantized model.
+        if self._evaluator.is_metric_mode():
```

Comment:
How will this approach be extended if somebody want to implement new ranking function? Why did you refuse from passing `ranking_fn` to `__init__`?

Reply:
Fixed